File: api/src/main/java/com/castlefrog/agl/AdversarialSimulator.java
package com.castlefrog.agl;

public abstract class AdversarialSimulator<S extends State<S>, A extends Action>
        extends AbstractSimulator<S, A> {
    protected static final int N_AGENTS = 2;
    protected static final int[] REWARDS_NEUTRAL = new int[] { 0, 0 };
    protected static final int[] REWARDS_BLACK_WINS = new int[] { 1, -1 };
    protected static final int[] REWARDS_WHITE_WINS = new int[] { -1, 1 };

    protected AdversarialSimulator() {}

    protected AdversarialSimulator(AdversarialSimulator<S, A> simulator) {
        super(simulator);
    }

    public int getNAgents() {
        return N_AGENTS;
    }
}

File: api/src/main/java/com/castlefrog/agl/AdversarialSimulator.kt
package com.castlefrog.agl

abstract class AdversarialSimulator<S : State<S>, A : Action<A>> : Simulator<S, A> {

    override val nAgents: Int = N_AGENTS

    companion object {
        val N_AGENTS = 2
        val REWARDS_NEUTRAL = intArrayOf(0, 0)
        val REWARDS_BLACK_WINS = intArrayOf(1, -1)
        val REWARDS_WHITE_WINS = intArrayOf(-1, 1)
    }
}
